### ISSUE TITLE
ReactiveCommand gives System.InvalidOperationException after binding 

### ISSUE THREAD

**Issue by goyzhang** [reactiveui/ReactiveUI#1676]
_We will close your issue without comment if you delete or do not fill out the issue checklist and provide ALL the requested information. This repository's issues are reserved for feature requests and bug reports. For support questions, please ask on StackOverflow: https://stackoverflow.com/questions/tagged/reactiveui_

_Please delete this note before pressing the submit button_

**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

After binding the command to , for instance, a button. Click the button will cause `System.InvalidOperationException` indicating that object owns by another thread right after the callback returns.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

in xaml:
`<Button Command="{Binding MyCommand}"></Button>`

in viewmodel:
        
```
public ReactiveCommand MyCommand { get; protected set; }
        public MainWindowViewModel()
        {
            MyCommand =ReactiveCommand.CreateFromTask(() =>
            {
                Console.WriteLine("foo"); //this works
                return Task.Run(() => {

                });
            });//cause exception right after return
        }
```

**What is the expected behavior?**

Works as any command. No exception.

**What is the motivation / use case for changing the behavior?**

**Which versions of ReactiveUI, and which platform / OS are affected by this issue? Did this work in previous versions of ReativeUI? Please also test with the [latest stable](https://www.nuget.org/profiles/reactiveui) and [development snapshot](https://reactiveui.net/docs/getting-started/installation/nuget-packages/development-package-feed)**
In v8.3.1 with reactive 3.3.1 & .net 4.7.1 WPF project
Same codes work fine in v7.4

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

**Comment by phenomx2** [comment]
Why are you using Task.Run(...) why not use await ?

**Comment by goyzhang** [comment]
@phenomx2  That makes no different. I just want to give it a quick demo and the API does not require that to be async/await pattern.

**Comment by phenomx2** [comment]
Can you share a small repo with the bug ? please

**Comment by goyzhang** [comment]
check this [repo](https://github.com/goyzhang/ReactiveUIDemoRepo.git)

**Comment by Fido789** [comment]
I can confirm this bug. Even empty command ReactiveCommand.Create(()=>{}) throws the exception.

**Comment by Qonstrukt** [comment]
ReactiveCommand without generic parameters should never be used. It's still there because of compatibility with ICommand. But when you define your property like this:
```
public ReactiveCommand<Unit, Unit> MyCommand { get; set;}
```
You'll see that it works. It's not clear at all though, and the docs use the wrong (old) syntax still sometimes. Everything always seems to work, until you start binding.

**Comment by glennawatson** [comment]
We probably should add a issue on the website project to upload the instances where this is wrong in the documentation I guess.

**Comment by goyzhang** [comment]
umm.In v7.4 it works fine though. Is it possible to make this to be a synax error so that it won't even pass the compiler?

**Comment by glennawatson** [comment]
Potentially, will investigate and see if I can make the change. Obviously will be breaking but will likely stop fairly subtle bugs like this one.

**Comment by worldbeater** [comment]
This happens 'cause ReactiveUI schedulers won't get properly initialized when one uses a XAML UI framework and don't have a platform-specific package installed. Shortly, `ReactiveUI` package should normally be installed into your .NET Standard assembly only. For other platforms use one of these packages: `ReactiveUI.WPF`, `ReactiveUI.XamForms`, `ReactiveUI.WinForms` or `Avalonia.ReactiveUI`. See <a href="https://reactiveui.net/docs/getting-started/installation/nuget-packages/">Installation/NuGet-Packages</a> for more info.

**Comment by glennawatson** [comment]
Reopening since using this as a tracker to remove the abstract ReactiveCommand class.

**Comment by glennawatson** [comment]
Fixed by #1836 where we removed the abstract base non-generic class.
